chatminer/chatparsers.py

```diff
@@ -240,6 +240,54 @@ def _parse_message(self, mess):
         return parsed_message
 
 
+class InstagramChatsParser(Parser):
+    def _read_file_into_list(self):
+        self._logger.info("Starting reading raw messages into memory...")
+        self.messages = []
+        with self._file.open(encoding="utf-8") as f:
+            messages_raw = reversed((json.load(f)["messages"]))
+
+        for line in messages_raw:
+            self.messages.append(line)
+        self._logger.info(
+            "Finished reading %i raw messages into memory.", len(self.messages)
+        )
+
+    def _parse_message(self, mess):
+        if "share" in mess:
+            body = "share"
+        elif "photos" in mess:
+            body = "photo"
+        elif "videos" in mess:
+            body = "video"
+        elif "audio_files" in mess:
+            body = "audio"
+        elif "content" in mess:
+            if any(
+                flag in mess["content"]
+                for flag in (
+                    " to your message",
+                    " in the poll.",
+                    " created a poll: ",
+                    " liked a message",
+                    "This poll is no longer available.",
+                    "'s poll has multiple updates.",
+                )
+            ):
+                return None
+            else:
+                body = mess["content"]
+        else:
+            return None
+
+        parsed_message = {
+            "datetime": datetime.datetime.fromtimestamp(mess["timestamp_ms"] / 1000),
+            "author": mess["sender_name"].encode("latin-1").decode("utf-8"),
+            "message": body.encode("latin-1").decode("utf-8"),
```

Comment:
Since Instagram's output displays Unicode characters as codes in the format of \u02ec, the output ends up being wrong. I found that encoding and decoding the message made it work properly.

Reply:
did you think of using `unicodedata.normalize('NFKD', mess["sender_name"])`?https://stackoverflow.com/questions/16467479/normalizing-unicode
Not an expert in unicode wrangling in python, but this seems to be more straight forward

Reply:
I might be misunderstanding what `normalize` does, but the problem isn't two characters being equivalent, it's that they're being interpreted strangely because Instagram encodes its Unicode characters in a strange way. I've done some testing and it seems like it takes the character's value in UTF-8 code (I'm going to use the 😀 character for examples) such as F0 9F 98 80 and adds \u00 to the start of each bit. I can't find any other mentions of this encoding online.

For example, the character 😀, which is formatted by Instagram into \u00f0\u009f\u0098\u0080, gets interpreted somewhere in the code as ð. The only way I found to fix it was to do the strange encode decode thing.

Reply:
I just tested `normalize` (after spending two hours researching unicode encoding 😀 even though I should have tried it first) and it didn't work.

Reply:
Alright! Let's stick with this, thank you for doing the extra mile here!